CrowsZA/functions/fn_surrenderChanceZeus.sqf

```diff
@@ -0,0 +1,81 @@
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fn_surrenderChanceZeus.sqf
+Parameters: pos, unit
+Return: none
+
+*///////////////////////////////////////////////
+params [["_pos",[0,0,0],[[]],3], ["_unit",objNull,[objNull]]];
+
+private _onConfirm =
+{
+	params ["_dialogResult","_in"];
+	_in params [["_pos",[0,0,0],[[]],3], "_unit"];
+	private "_units";
+
+	if(isNull _unit) then {
+		_dialogResult params [
+			["_side", east, [east]],
+			["_confrontationRadius", 15, [15]],
+			["_surrenderChance", 0.33, [0.33]],
+			["_holdFire", true, [true]],
+			["_hesitation", 0, [0]]
+		];
+		_units = units _side;
+		[_units, _confrontationRadius, _surrenderChance, _holdFire, _hesitation] spawn {
+			params["_units", "_confrontationRadius", "_surrenderChance", "_holdFire", "_hesitation"];
+			{
+				if(!(isPlayer _x)) then {
+					[_x, _confrontationRadius, _surrenderChance, _holdFire, _hesitation] call crowsZA_fnc_surrenderChance;
+					sleep 0.01;
+				};
+			} foreach _units;
+		};
+	} else {
+		_dialogResult params [
+			["_group", true, [true]],
+			["_confrontationRadius", 15, [15]],
+			["_surrenderChance", 0.33, [0.33]],
+			["_holdFire", true, [true]],
+			["_hesitation", 0, [0]]
+		];
+		_units = [[_unit], units group _unit] select _group;
+		[_units, _confrontationRadius, _surrenderChance, _holdFire, _hesitation] spawn {
+			params["_units", "_confrontationRadius", "_surrenderChance", "_holdFire", "_hesitation"];
+			{
+				if(!(isPlayer _x)) then {
+					[_x, _confrontationRadius, _surrenderChance, _holdFire, _hesitation] call crowsZA_fnc_surrenderChance;
```

Comment:
Same as above point about scheduled and unscheduled space. You can read a bit about it here: https://ace3.acemod.org/wiki/development/coding-guidelines.html#72-scheduled-vs-unscheduled